pkg/agent/containerd/containerd.go

```diff
@@ -74,28 +71,15 @@ func Run(ctx context.Context, cfg *config.Node) error {
 	}()
 
 	for {
-		addr, dialer, err := util.GetAddressAndDialer("unix://" + cfg.Containerd.Address)
-		if err != nil {
-			time.Sleep(1 * time.Second)
-			continue
-		}
-
-		conn, err := grpc.Dial(addr, grpc.WithInsecure(), grpc.WithTimeout(3*time.Second), grpc.WithContextDialer(dialer), grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(maxMsgSize)))
-		if err != nil {
-			time.Sleep(1 * time.Second)
-			continue
-		}
-
-		c := runtimeapi.NewRuntimeServiceClient(conn)
-
-		_, err = c.Version(ctx, &runtimeapi.VersionRequest{
-			Version: "0.1.0",
-		})
+		client, err := containerd.New(cfg.Containerd.Address)
 		if err == nil {
-			conn.Close()
-			break
+			_, err = client.Version(ctx)
```

Comment:
@ibuildthecloud do we want to assert that containerd is running or that the CRI has been fully initialized? With your change, code relying on this check to assert that CRI is ready would fail when talking to it if CRI failed to recover at start-up. See https://github.com/containerd/cri/blob/release/1.3/pkg/server/service.go#L193

Reply:
This could also setup the client in a race with CRI taking a while to recover (which can be non-trivial for large installations.)